src/widgets/dialogs/UserInfoPopup.cpp

```diff
@@ -426,12 +426,19 @@ void UserInfoPopup::installEvents()
             switch (newState)
             {
                 case Qt::CheckState::Unchecked: {
-                    this->ui_.follow->setEnabled(false);
-                    getHelix()->unfollowUser(currentUser->getUserId(),
-                                             this->userId_,
-                                             reenableFollowCheckbox, [] {
-                                                 //
-                                             });
+                    auto box = new QMessageBox(
+                        QMessageBox::Information, "Chatterino 2",
+                        "Unfollow user?", QMessageBox::Yes | QMessageBox::No);
+                    box->setAttribute(Qt::WA_DeleteOnClose);
+                    if (box->exec() == QMessageBox::Yes)
```

Comment:
Not a QT guru but I suspect blocking the UI with this call is your cause. Try using `QObject::connect` to connect to the `clicked` signal. [See here](https://github.com/Chatterino/chatterino2/blob/cb56e414a4fd1b90b25bc03ef5c7de2daeb9ae3d/src/widgets/dialogs/UpdateDialog.cpp#L28) for an example.

QT devs correct me if I'm wrong.

Reply:
This applies to all 4 usages of `exec()`